pkg/sql/backfill.go

```diff
@@ -1023,6 +999,56 @@ func (sc *SchemaChanger) distIndexBackfill(
 		return nil
 	}
 
+	writeAsOf := sc.job.Details().(jobspb.SchemaChangeDetails).WriteTimestamp
+	if writeAsOf.IsEmpty() {
+		if err := sc.job.RunningStatus(ctx, nil /* txn */, func(_ context.Context, _ jobspb.Details) (jobs.RunningStatus, error) {
+			return jobs.RunningStatus("scanning target index for in-progress transactions"), nil
+		}); err != nil {
+			return errors.Wrapf(err, "failed to update running status of job %d", errors.Safe(sc.job.ID()))
+		}
+		writeAsOf = sc.clock.Now()
+		log.Infof(ctx, "starting scan of target index as of %v...", writeAsOf)
+		// Index backfilling ingests SSTs that don't play nicely with running txns
+		// since they just add their keys blindly. Running a Scan of the target
+		// spans at the time the SSTs' keys will be written will calcify history up
+		// to then since the scan will resolve intents and populate tscache to keep
+		// anything else from sneaking under us. Since these are new indexes, these
+		// spans should be essentially empty, so this should be a pretty quick and
+		// cheap scan.
+		const pageSize = 10000
```

Comment:
For my own education: how was this batch size chosen? 10k seems to be the norm. I'm guessing it doesn't matter too much as long as it's large enough to amortize the overhead but not too large?
I realize this code was already there and only shows up in the diff because it was wrapped in an `if`.